cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -0,0 +1,166 @@
+package bicep
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/azure"
+	"github.com/azure/azure-dev/cli/azd/pkg/azureutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"golang.org/x/exp/slices"
+
+	. "github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
+)
+
+func (p *BicepProvider) promptForParameter(ctx context.Context, key string, param azure.ArmTemplateParameter) (any, error) {
```

Comment:
> It might be interesting to split this out to be reused in other scenarios (or happy to hold off on this till we get to this point)

I do think this will come as well but I'm going to hold off for now.